test/integration/test___main__.py

```diff
@@ -42,30 +41,26 @@ def test_main_bad_private_data_dir():
     cmdline('run', tmpfile, '-p', 'fake')
 
     try:
-        with raises(OSError):
+        with pytest.raises(OSError):
             main()
     finally:
         os.remove(tmpfile)
 
 
-def run_role(options, private_data_dir=None, expected_rc=0):
-    try:
```

Comment:
Removing this outer `try` revealed test failures that were being swallowed by the `finally` statement. It's related to #825. Can't get rid of that code fast enough.